### PR TITLE
Update protobuf to 3.6.1

### PR DESCRIPTION
**Pull request by rbsheth**
@ruslo Lots of the cherry-picked commits for the 3.5.2 Hunterization are already in the 3.6 branch, so I simply added only the hunterization commit via a rebase.

Needs a new branch in hunter-packages/protobuf, probably hunter-3.6.x or hunter-3.6.1.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -39,10 +39,9 @@ COPTS = select({
     ],
 })
 
-config_setting(
-    name = "msvc",
-    values = { "compiler": "msvc-cl" },
-)
+load(":compiler_config_setting.bzl", "create_compiler_config_setting")
```
Comment (ruslo) [inline BUILD]: > I simply added only the hunterization commit via a rebase

I think you pull changes that we don't need. I'm expecting only CMake code touched, see previous hunterization:
* https://github.com/hunter-packages/protobuf/pull/7/files

**Review by ruslo — CHANGES_REQUESTED** [review]
> Needs a new branch in hunter-packages/protobuf, probably hunter-3.6.x or hunter-3.6.1.

I've created branch `hunter-3.6.1` from upstream tag `v3.6.1`.

**Comment by rbsheth** [comment]
@ruslo there are a number of backported fixes to the branch post 3.6.1 release: https://github.com/protocolbuffers/protobuf/commits/3.6.x

Some of them are quite important, such as Python 3.7 compatibility. Should we make a new version for this? Something like 3.6.1-b829ff2 or 3.6.x-b829ff2?

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
> Some of them are quite important, such as Python 3.7 compatibility

Okay, I see

> Something like 3.6.1-b829ff2 or 3.6.x-b829ff2?

Yes, branch hunter-3.6.x-b829ff2 created

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/protobuf/releases/tag/v3.6.x-b829ff2-p0

**Pull request merged**